gcloud-java-compute/src/main/java/com/google/gcloud/compute/DeprecationStatus.java

```diff
@@ -64,39 +67,192 @@
+     */
+    public Builder<T> obsolete(long obsolete) {
+      this.obsolete = TIMESTAMP_FORMATTER.print(obsolete);
+      return this;
+    }
+
+    /**
+     * Sets the identity of the suggested replacement for a deprecated resource. The suggested
+     * replacement resource must be the same kind of resource as the deprecated resource.
+     */
+    public Builder<T> replacement(T replacement) {
+      this.replacement = replacement;
+      return this;
+    }
+
+    /**
+     * Sets the status of the deprecated resource.
+     */
+    public Builder<T> status(Status status) {
+      this.status = checkNotNull(status);
+      return this;
+    }
+
+    /**
+     * Creates a {@code DeprecationStatus} object.
+     */
+    public DeprecationStatus<T> build() {
+      return new DeprecationStatus<T>(this);
+    }
+  }
+
+  DeprecationStatus(Builder<T> builder) {
+    this.deleted = builder.deleted;
+    this.deprecated = builder.deprecated;
+    this.obsolete = builder.obsolete;
+    this.replacement = builder.replacement;
+    this.status = checkNotNull(builder.status);
   }
 
   /**
    * Returns the timestamp on or after which the deprecation state of this resource will be changed
-   * to {@link Status#DELETED}. In milliseconds since epoch.
+   * to {@link Status#DELETED}. Returns {@code null} if not set. This value should be in RFC3339
+   * format.
+   *
+   * @see <a href="http://www.ietf.org/rfc/rfc3339.txt">RFC3339</a>
    */
-  public Long deleted() {
+  // todo(mziccard): remove this method if #732 is closed
+  public String deleted() {
     return deleted;
   }
 
   /**
    * Returns the timestamp on or after which the deprecation state of this resource will be changed
-   * to {@link Status#DEPRECATED}. In milliseconds since epoch.
+   * to {@link Status#DEPRECATED}. Returns {@code null} if not set. This value should be in RFC3339
+   * format.
+   *
+   * @see <a href="http://www.ietf.org/rfc/rfc3339.txt">RFC3339</a>
    */
-  public Long deprecated() {
+  // todo(mziccard): remove this method if #732 is closed
+  public String deprecated() {
     return deprecated;
   }
 
   /**
    * Returns the timestamp on or after which the deprecation state of this resource will be changed
-   * to {@link Status#OBSOLETE}. In milliseconds since epoch.
+   * to {@link Status#OBSOLETE}. Returns {@code null} if not set. This value should be in RFC3339
+   * format.
+   *
+   * @see <a href="http://www.ietf.org/rfc/rfc3339.txt">RFC3339</a>
    */
-  public Long obsolete() {
+  // todo(mziccard): remove this method if #732 is closed
+  public String obsolete() {
     return obsolete;
   }
 
   /**
+   * Returns the timestamp (in milliseconds since epoch) on or after which the deprecation state of
+   * this resource will be changed to {@link Status#DELETED}. Returns {@code null} if not set.
+   *
+   * @throws IllegalArgumentException if {@link #deleted()} is not a valid date, time or datetime
```

Comment:
Yes, I realized that. I think it is strange to throw an `IllegalArgumentException` for a method that does not accept an argument and in such cases `IllegalStateException` is commonly used. It is a nit, but catch and re-throw would be my preference but will not object if you think otherwise.